cpmpy/solvers/pysat.py

```diff
@@ -89,96 +92,50 @@ def __init__(self, cpm_model=None, solver=None):
             raise Exception("CPM_pysat: Install the python 'python-sat' package to use this solver interface (NOT the 'pysat' package!)")
         if cpm_model and cpm_model.objective is not None:
             raise Exception("CPM_pysat: only satisfaction, does not support an objective function")
+
         from pysat.formula import IDPool
         from pysat.solvers import Solver
 
-        super().__init__(cpm_model, solver)
-
         # determine solvername, set cpmpy name
         solvername = solver
         if solver is None or solvername == 'pysat':
             # default solver
             solvername = "glucose4" # something recent...
         elif solvername.startswith('pysat:'):
             solvername = solvername[6:] # strip 'pysat:'
-        self.name = "pysat:"+solvername
 
-        # ID pool of variables
+        # Create solver specific objects
         self.pysat_vpool = IDPool()
-
-        if cpm_model is None:
-            self.user_vars = []
-            from pysat.formula import CNF
-            cnf = CNF()
-        else:
-            # store original vars
-            self.user_vars = get_variables_model(cpm_model)
-
-            # create constraint model (list of clauses)
-            cnf = self.make_cnf(cpm_model)
-
-        # create the solver instance
+        cnf = pysat.formula.CNF()
         self.pysat_solver = Solver(bootstrap_with=cnf.clauses, use_timer=True, name=solvername)
```

Comment:
no need to create an empty CNF to then bootstrap with it, I think?

Reply:
True, this is a leftover artefact of the make_model strategy we had.